src/Handlers/SaveScreenOptionsHandler.php

```diff
@@ -0,0 +1,182 @@
+     *
+     * @since [*next-version*]
+     *
+     * @param string                     $screenOptionsKey    Option key name to save screen statuses config.
+     * @param array|stdClass|Traversable $screenOptionsFields List of fields in screen options.
+     * @param SimpleCacheInterface       $screenOptionsCache  The screen options cache.
+     */
+    public function __construct($screenOptionsKey, $screenOptionsFields, $screenOptionsCache)
+    {
+        $this->screenOptionsKey    = $screenOptionsKey;
+        $this->screenOptionsFields = $screenOptionsFields;
+        $this->screenOptionsCache  = $screenOptionsCache;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        /* @var $event EventInterface */
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        if (!$user = wp_get_current_user()) {
+            return;
+        }
+
+        $this->_handle($user->ID);
+    }
+
+    /**
+     * Save screen options for user.
+     *
+     * @since [*next-version*]
+     *
+     * @param int $userId User identifier.
+     */
+    protected function _handle($userId)
+    {
+        $data    = $this->_getRequestData();
+        $options = [];
+        foreach ($this->screenOptionsFields as $key) {
+            if (array_key_exists($key, $data)) {
+                $options[$key] = $data[$key];
+            }
+        }
+        $this->_setScreenOptions($userId, $options);
+    }
+
+    /**
+     * Save screen options in per-user options.
+     *
+     * @since [*next-version*]
+     *
+     * @param int            $userId  User identifier.
+     * @param array|stdClass $options Screen options to save.
+     */
+    protected function _setScreenOptions($userId, $options)
+    {
+        $userIdKey = $this->_normalizeKey($userId);
+        $key       = $this->screenOptionsKey;
+
+        $this->_updateUserOption($userId, $key, json_encode($options));
+
+        if ($this->screenOptionsCache->has($userIdKey)) {
+            $this->screenOptionsCache->delete($userIdKey);
+        }
+    }
+
+    /**
+     * Get data from request.
+     *
+     * @since [*next-version*]
+     *
+     * @return array|mixed|object Request data.
+     */
+    protected function _getRequestData()
+    {
+        return json_decode(file_get_contents('php://input'), true);
+    }
+
+    /**
+     * Update user option.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|float|string|Stringable $userId User ID.
+     * @param string|Stringable           $key    User option key.
+     * @param mixed                       $value  User option value.
+     *
+     * @return bool|int User meta ID if the option didn't exist, true on successful update, false on failure.
```

Comment:
This is very specific to the implementation detail, in particular - to the `update_user_option()` function. A function that is not a getter must do what it's supposed to do, or throw an exception. So, update the option, or if that fails - throw a `RuntimeException`.